sound/soc/intel/boards/sof_maxim_common.h

```diff
@@ -0,0 +1,88 @@
+/* SPDX-License-Identifier: GPL-2.0 */
+/*
+ * Copyright(c) 2020 Intel Corporation.
+ */
+
+/*
+ * This file defines data structures used in Machine Driver for Intel
+ * platforms with Maxim Codecs.
+ */
+#ifndef __SOF_MAXIM_COMMON_H
+#define __SOF_MAXIM_COMMON_H
+
+#define MAX98373_CODEC_DAI	"max98373-aif1"
+#define MAX98373_DEV0_NAME	"i2c-MX98373:00"
+#define MAX98373_DEV1_NAME	"i2c-MX98373:01"
+
+static int max98373_hw_params(struct snd_pcm_substream *substream,
+	struct snd_pcm_hw_params *params)
+{
+	struct snd_soc_pcm_runtime *runtime = substream->private_data;
+	int j;
+
+	for (j = 0; j < runtime->num_codecs; j++) {
+		struct snd_soc_dai *codec_dai = runtime->codec_dais[j];
+
+		if (!strcmp(codec_dai->component->name, MAX98373_DEV0_NAME)) {
+			/* DEV0 tdm slot configuration */
+			snd_soc_dai_set_tdm_slot(codec_dai, 0x30, 3, 8, 16);
+		}
+		if (!strcmp(codec_dai->component->name, MAX98373_DEV1_NAME)) {
+			/* DEV1 tdm slot configuration */
+			snd_soc_dai_set_tdm_slot(codec_dai, 0xC0, 3, 8, 16);
```

Comment:
why do you need 8 slots, looks like you could use only 4?

Reply:
We wan to keep 8-TDM slots for the support if 4-amp's connected. So, same can be re-used.
Also, one advantage from Codec side is:
With 8-slots, amp had been verified well in KBL products.. so want to use same working configuration (which seems to be good for TGL too).

Reply:
ok, but then look at sof_da7219_max98373, the code uses 4 slots.

Reply:
I cross checked, in that program there is no plan to go for 4-amp's.. so 4-slots is good for them. With these codec combinations in current projects, there is future plan for 4-amp's: so: I am keeping as 8-slots.